### PR TITLE
[Arm64] Add corefx_test_exclusions.txt

### PR DESCRIPTION
**Pull request by sdmaclea**
@BruceForstall 

This was a list of the tests which are failing.

### CODE REVIEW

File: tests/arm64/corefx_test_exclusions.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
+Invariant.Tests
+System.Diagnostics.Process.Tests
+System.IO.FileSystem.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: This test takes ~ 30 minutes due to 6 tests which take ~5minute & 4 tests which take ~30s.
Also fails intermittently

File: tests/arm64/corefx_test_exclusions.txt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+System.Linq.Expressions.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: This test takes ~10 minutes

File: tests/arm64/corefx_test_exclusions.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+System.Net.Requests.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: This test has a dependency on running w/o a firewall.

File: tests/arm64/corefx_test_exclusions.txt [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+System.Reflection.Emit.ILGeneration.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: P/Invoke Calli is broken?

File: tests/arm64/corefx_test_exclusions.txt [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+System.Security.Cryptography.Algorithms.Tests
+System.Security.Cryptography.Encoding.Tests
+System.Security.Cryptography.OpenSsl.Tests
+System.Security.Cryptography.Primitives.Tests
+System.Text.Encoding.CodePages.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Fix was merged to corefx this week

File: tests/arm64/corefx_test_exclusions.txt [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+System.Threading.Channels.Tests
+System.Threading.Channels.Tests
+System.Threading.Tasks.Dataflow.Tests
+System.Threading.Tasks.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Fix merged to corefx this week

File: tests/arm64/corefx_test_exclusions.txt [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+System.IO.Pipes.Tests
+System.ValueTuple.Tests
+System.Xml.RW.XmlConvert.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: This test takes ~5 minutes

File: tests/arm64/corefx_test_exclusions.txt [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+System.Security.Cryptography.Encoding.Tests
+System.Security.Cryptography.OpenSsl.Tests
+System.Security.Cryptography.Primitives.Tests
+System.Text.Encoding.CodePages.Tests
+System.Threading.Channels.Tests
+System.Threading.Channels.Tests
+System.Threading.Tasks.Dataflow.Tests
+System.Threading.Tasks.Tests
+System.IO.Pipes.Tests
+System.ValueTuple.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Known codegen issue

File: tests/arm64/corefx_test_exclusions.txt [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+System.Security.Cryptography.Algorithms.Tests
+System.Security.Cryptography.Encoding.Tests
+System.Security.Cryptography.OpenSsl.Tests
+System.Security.Cryptography.Primitives.Tests
+System.Text.Encoding.CodePages.Tests
+System.Threading.Channels.Tests
+System.Threading.Channels.Tests
+System.Threading.Tasks.Dataflow.Tests
+System.Threading.Tasks.Tests
+System.IO.Pipes.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Fix merged to corefx this week

File: tests/arm64/corefx_test_exclusions.txt [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+System.Net.Requests.Tests
+System.Reflection.Emit.ILGeneration.Tests
+System.Security.Cryptography.Algorithms.Tests
+System.Security.Cryptography.Encoding.Tests
+System.Security.Cryptography.OpenSsl.Tests
+System.Security.Cryptography.Primitives.Tests
+System.Text.Encoding.CodePages.Tests
+System.Threading.Channels.Tests
+System.Threading.Channels.Tests
+System.Threading.Tasks.Dataflow.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: GC and/or MemoryBarrier issues

File: tests/arm64/corefx_test_exclusions.txt (hunk 5 above)
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: GC and/or MemoryBarrier issues

File: tests/arm64/corefx_test_exclusions.txt (hunk 3 above)
Comment (sdmaclea) [reply tests/arm64/corefx_test_exclusions.txt]: Also broken with new regression.

File: tests/arm64/corefx_test_exclusions.txt [hunk 11]
```diff
@@ -0,0 +1,26 @@
+Invariant.Tests
+System.Diagnostics.Process.Tests
+System.IO.FileSystem.Tests
+System.Linq.Expressions.Tests
+System.Net.Requests.Tests
+System.Reflection.Emit.ILGeneration.Tests
+System.Security.Cryptography.Algorithms.Tests
+System.Security.Cryptography.Encoding.Tests
+System.Security.Cryptography.OpenSsl.Tests
+System.Security.Cryptography.Primitives.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Above failures likely due to GC and/or MemoryBarrier issues

File: tests/arm64/corefx_test_exclusions.txt [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+System.Text.Encoding.CodePages.Tests
+System.Threading.Channels.Tests
+System.Threading.Tasks.Dataflow.Tests
+System.Threading.Tasks.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Above failures likely due to GC and/or MemoryBarrier issues

File: tests/arm64/corefx_test_exclusions.txt [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+System.Threading.Thread.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Fix merged to corefx this week

File: tests/arm64/corefx_test_exclusions.txt [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+System.IO.Pipes.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Fix merged to corefx this week.

File: tests/arm64/corefx_test_exclusions.txt [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+System.ValueTuple.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Known codegen issue

File: tests/arm64/corefx_test_exclusions.txt [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+System.Xml.RW.XmlConvert.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: Takes ~5 minutes to run

File: tests/arm64/corefx_test_exclusions.txt [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+System.Net.HttpListener.Tests
+System.Net.Mail.Functional.Tests
+System.Net.NetworkInformation.Functional.Tests
+System.Net.Requests.Tests
+System.Net.Security.Tests
+System.Net.Sockets.Tests
+System.Net.WebClient.Tests
+System.Net.WebSockets.Client.Tests
```
Comment (sdmaclea) [inline tests/arm64/corefx_test_exclusions.txt]: New regression observed 4/3

**Comment by BruceForstall** [comment]
Looks like great progress. The Windows ARM corefx tests have not been watched, so there are lots of failures currently. E.g., "Microsoft.Win32.Registry.Tests" looks to have a few failures just due to directory names including architecture.

File: tests/arm64/corefx_test_exclusions.txt (hunk 11 above)
Comment (sdmaclea) [reply tests/arm64/corefx_test_exclusions.txt]: These were an issue with JIT_MemSet

File: tests/arm64/corefx_test_exclusions.txt (hunk 12 above)
Comment (sdmaclea) [reply tests/arm64/corefx_test_exclusions.txt]: These were an issue with the ThreadPool WorkStealing Queue

File: tests/arm64/corefx_test_exclusions.txt (hunk 17 above)
Comment (sdmaclea) [reply tests/arm64/corefx_test_exclusions.txt]: Fixed in tip

**Comment by BruceForstall** [comment]
@sdmaclea Do you want me to merge this? It's not references by anyone, since there is no arm64 corefx netci.groovy support yet that references it.

**Comment by sdmaclea** [comment]
Up to you .  You were discussing enabling corefx on linux.  I was providing this to unblock that activity.

**Comment by BruceForstall** [comment]
I'll go ahead and merge it, then. If you (or others) make a disabled test run on arm64, a PR can remove the test from the list.

**Pull request merged**